### PR TITLE
gitleaks 2.0.0

### PR DESCRIPTION
**Pull request by chenrui333**

### CODE REVIEW

**Comment by chenrui333** [comment]
work on it!

**Comment by chenrui333** [comment]
```
23:41:21 cp -pR /private/tmp/d20190514-94987-1l030bm/gitleaks-2.0.0/. /private/tmp/gitleaks-20190514-94987-4wup1w/gitleaks-2.0.0
23:41:21 chmod -Rf +w /private/tmp/d20190514-94987-1l030bm
23:41:21 Error: An exception occurred within a child process:
23:41:21   Errno::EINVAL: Invalid argument @ rb_file_s_rename - (/private/tmp/gitleaks-20190514-94987-4wup1w/gitleaks-2.0.0/src, /private/tmp/gitleaks-20190514-94987-4wup1w/gitleaks-2.0.0/src/github.com/zricethezav/gitleaks/src)
```

**Comment by chenrui333** [comment]
Already been upgraded in https://github.com/Homebrew/homebrew-core/pull/39581

**Comment by chenrui333** [comment]
circle back to try fixing the upgrade issue. 
cc @zricethezav

**Comment by zricethezav** [comment]
@chenrui333 gitleaks 2.0.0 does not depend on dep anymore, instead it uses go mod
see the closed PR here: https://github.com/Homebrew/homebrew-core/pull/39894/files

**Comment by chenrui333** [comment]
yeah, indeed. I updated in my local (just pushed now).

I am still seeing this error:
```
Errno::EINVAL: Invalid argument @ rb_file_s_rename - (/private/tmp/gitleaks-20190516-99060-gpfsji/gitleaks-2.0.0/src, /private/tmp/gitleaks-20190516-99060-gpfsji/gitleaks-2.0.0/src/github.com/zricethezav/gitleaks/src)
```

**Comment by zbeekman** [comment]
Can someone push the required updates to this PR? Or open a new PR? Between 39581 and this, it should work, no? Can you mimic what was done in the other PR but based on a recent master?

**Comment by chenrui333** [comment]
@zbeekman I still have install issue for this formula in my local:
```
tar xof /Users/rchen/Library/Caches/Homebrew/downloads/b784582115bee6d2bb429a98c596fa68fc537240836cdce63005e40d6620d7dd--gitleaks-2.0.0.tar.gz -C /private/tmp/d20190522-52483-1fdm3od
cp -pR /private/tmp/d20190522-52483-1fdm3od/gitleaks-2.0.0/. /private/tmp/gitleaks-20190522-52483-8yztc0/gitleaks-2.0.0
chmod -Rf +w /private/tmp/d20190522-52483-1fdm3od
Error: An exception occurred within a child process:
  Errno::EINVAL: Invalid argument @ rb_file_s_rename - (/private/tmp/gitleaks-20190522-52483-8yztc0/gitleaks-2.0.0/src, /private/tmp/gitleaks-20190522-52483-8yztc0/gitleaks-2.0.0/src/github.com/zricethezav/gitleaks/src)
```

**Comment by chenrui333** [comment]
@zricethezav feel free to review the updated PR.

File: Formula/gitleaks.rb [hunk 1]
```diff
@@ -1,8 +1,9 @@
 class Gitleaks < Formula
   desc "Audit git repos for secrets"
   homepage "https://github.com/zricethezav/gitleaks"
-  url "https://github.com/zricethezav/gitleaks/archive/v1.24.0.tar.gz"
-  sha256 "6ba812be47976ca49bc2f5ab888c44ef41b824dd20fa9be5687f4ff6d185c2b1"
+  url "https://github.com/zricethezav/gitleaks/archive/v2.0.0.tar.gz"
+  sha256 "85a5c98dedeb4e85e07eb18247b63318aa266ef3046c2022eac949cc6f254da0"
+  head "https://github.com/zricethezav/gitleaks.git"
```
Comment (zbeekman) [inline Formula/gitleaks.rb]: Also remove head! You need to stop adding this everywhere. It should only be added if there is a good reason. E.g., upstream wants to do nightly testing with homebrew. Or you personally need and use it, and have tested it.

**Review by zbeekman — CHANGES_REQUESTED** [review]

File: Formula/gitleaks.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/gitleaks.rb]: yeah, sorry about the `head` issue spread, not quite understand the usage. I will remove it.

**Comment by chenrui333** [comment]
@zbeekman
`Errno::EINVAL: Invalid argument @ rb_file_s_rename`, I could not resolve this issue.

**Comment by zricethezav** [comment]
This can be closed, I opened a new PR that is working: https://github.com/Homebrew/homebrew-core/pull/40734 cc @chenrui333 @zbeekman

**Comment by chenrui333** [comment]
@zricethezav Cool, sounds good!! Thanks!

**Pull request closed without merging**